packages/zoe/src/zoe.js

```diff
@@ -24,10 +25,215 @@ import { areRightsConserved } from './rightsConservation';
+ * By having a reference to Zoe, a user can get the `inviteIssuer`
+ * and thus validate any `invite` they receive from someone else. The
+ * mint associated with the inviteIssuer creates the ERTP payments
+ * that represent the right to interact with a smart contract in
+ * particular ways.
+ *
+ * @property {(code: string, moduleFormat: string) => InstallationHandle} install
+ * Create an installation by safely evaluating the code and
+ * registering it with Zoe. Returns an installationHandle.
+ *
+ * @property {(installationHandle: InstallationHandle, issuerKeywordRecord: IssuerKeywordRecord, terms: object) => Invite} makeInstance
+ * Zoe is long-lived. We can use Zoe to create smart contract
+ * instances by specifying a particular contract installation to
+ * use, as well as the `issuerKeywordRecord` and `terms` of the contract. The
+ * `issuerKeywordRecord` is a record mapping string names (keywords) to issuers,
+ * such as `{ Asset: simoleanIssuer}`. (Note that the keywords must
+ * begin with a capital letter and must be ASCII.) Parties to the
+ * contract will use the keywords to index their proposal and
+ * their payments.
+ *
+ * The payout users receive from Zoe will be in the form of an object
+ * with keywords as keys. Terms are the arguments to the contract,
+ * such as the number of bids an auction will wait for before closing.
+ * Terms are up to the discretion of the smart contract. We get back
+ * an invite (an ERTP payment) to participate in the contract.
+ *
+ * TODO: property {(instanceHandle: InstanceHandle) => InstanceRecord} getInstanceRecord
+ * Credibly get information about the instance (such as the installation
+ * and terms used).
+ *
+ * @property {(invite: Invite, proposal: Proposal, payments: PaymentKeywordRecord) => Promise<SeatAndPayout>} redeem
+ * To redeem an invite, the user normally provides a proposal (their rules for the
+ * offer) as well as payments to be escrowed by Zoe.  If either the proposal or payments
+ * would be empty, indicate this by omitting that argument or passing undefined, rather
+ * than passing an empty record.
+ *
+ * The proposal has three parts: `want` and `give` are used
+ * by Zoe to enforce offer safety, and `exit` is used to specify
+ * the extent of payout liveness that Zoe can guarantee.
+ * `want` and `give` are objects with keywords as keys and amounts
+ * as values. `payments` is a record with keywords as keys,
+ * and the values are the actual payments to be escrowed. A payment
+ * is expected for every rule under `give`.
+ *
+ * @property {(installationHandle: InstallationHandle) => InstallationRecord} getInstallation
+ *
+ * @typedef {Object} SeatAndPayout This is returned by a call to `redeem` on Zoe.
+ *
+ * @property {Object} seat An arbitrary object whose methods allow the user to take
+ * certain actions in a contract. The payout is a promise that resolves
+ * to an object which has keywords as keys and promises for payments
```

Comment:
The payout promise part should be removed, since it's duplicated below

Reply:
I'll leave this for @Chris-Hibbert.